blazor/datagrid/custom-toolbar.md

```diff
@@ -0,0 +1,462 @@
+        </ToolbarItems>
+    </SfToolbar>
+    <GridColumns>
+        <GridColumn Field=@nameof(OrderData.OrderID) HeaderText="Order ID" IsPrimaryKey="true" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.CustomerID) HeaderText="Customer Name" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.OrderDate) HeaderText=" Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Right" Width="130"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+    </GridColumns>
+</SfGrid>
+@code {
+    private SfGrid<OrderData> Grid;
+    public List<OrderData> Orders { get; set; }
+    
+    protected override void OnInitialized()
+    {
+        Orders = OrderData.GetAllRecords();
+    }
+    public void AddButton()
+    {
+        this.Grid.AddRecordAsync();
+    }
+    public void DeleteButton()
+    {
+        this.Grid.DeleteRecordAsync();
+    }
+}
+<style>
+    .image {
+        height: 50px;
+    }
+</style>
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+    public class OrderData
+    {
+        public static List<OrderData> Orders = new List<OrderData>();
+        public OrderData()
+        {
+
+        }
+        public OrderData(int? OrderID, string CustomerID, DateTime? OrderDate, double Freight)
+        {
+            this.OrderID = OrderID;
+            this.CustomerID = CustomerID;
+            this.OrderDate = OrderDate;
+            this.Freight = Freight;
+        }
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+                int code = 10;
+                for (int? i = 1; i < 2; i++)
+                {
+                    Orders.Add(new OrderData(1, "Nancy", new DateTime(1993, 09, 15), 98));
+                    Orders.Add(new OrderData(2, "Andrew", new DateTime(1997, 06, 01), 46));
+                    Orders.Add(new OrderData(3, "Steven", new DateTime(2000, 04, 04), 56));
+                    Orders.Add(new OrderData(4, "Margaret", new DateTime(1895, 11, 11), 74));
+                    Orders.Add(new OrderData(5, "Janet", new DateTime(2001, 08, 04), 83));
+                    Orders.Add(new OrderData(6, "Andrew", new DateTime(2022, 04, 09), 51));
+                    Orders.Add(new OrderData(7, "Nancy", new DateTime(2023, 06, 06), 23));
+                    Orders.Add(new OrderData(8, "Margaret", new DateTime(2011, 12, 30), 87));
+                    Orders.Add(new OrderData(9, "Janet", new DateTime(2012, 07, 07), 34));
+                    code += 5;
+                }
+            }
+            return Orders;
+        }
+        
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public DateTime? OrderDate { get; set; }
+        public double Freight { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
```

Comment:
playground sample is missing

Reply:
not able to use image files in PlayGround. So we aren't provide the preview sample.